resources/ant.py

```diff
@@ -94,5 +97,33 @@ def find_food(self, food_position):
             return (ant_x, ant_y)
         return None
     
-    def carry_food(self):
-        pass
+
+    
+    def carry_food(self, food):
+        """
+        Enables the ant to take food from food_source if certain conditions are met 
+        and updates its pheromonestatus and the amount_of_food left at the food-source
+
+        Args:
+            food (Object): The food_source where the ant is trying to take food from
+        
+        Return:
+            None: The method changes the state of the Ant and the food_source directly 
+        """
+        # if conditions are matched, switch state of ant to carryfood
+        if self.pheromone_status == -1 and self.find_food(food.coordiantes) and food.amount_of_food > 0:
+            self.carrying_food = True
+
+            # subtracts amount to carry or whatevers left 
+            amount_taken = min(food.amount_of_food, self.amount_to_carry)
+            food.amount_of_food -= amount_taken
+
+            #call switch pheromone method
+            self.switch_pheromone()
+
+        #add blank for if food source is empty
+        elif food.amount_of_food == 0:
```

Comment:
This is unnecessary because if the if condition returns false, nothing further is done anyway! Please either delete it or explain the reasoning behind it. 😄